### PR TITLE
Fix Jenkins plugin not working at all

### PR DESCRIPTION
**Pull request by timja**
Signed-off-by: Tim Jacomb <timjacomb1@gmail.com>

## Hey, I just made a Pull Request!

Fixes https://github.com/backstage/backstage/issues/13692

not sure what caused this, some bundling change maybe, I can't see anything related from a quick look.
I've tested this locally and it fixes the issue

I think this is only an issue in the weekly releases (.next) so unsure if you would normally changeset it but I've included one.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- ~[ ] Added or updated documentation~
- [ ] Tests for new functionality and regression tests for bug fixes
- ~[ ] Screenshots attached (for UI changes)~
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/jenkins-backend/src/service/jenkinsApi.ts [hunk 1]
```diff
@@ -177,12 +177,12 @@ export class JenkinsApiImpl {
 
   private static async getClient(jenkinsInfo: JenkinsInfo) {
     // The typings for the jenkins library are out of date so just cast to any
-    return jenkins({
+    return new (Jenkins as any)({
```
Comment (benjdlambert) [inline plugins/jenkins-backend/src/service/jenkinsApi.ts:180]: Do we really need the `as any` still here? We've got `@types/jenkins` installed, so maybe they need updating?

**Review by benjdlambert — CHANGES_REQUESTED** [review]

File: plugins/jenkins-backend/src/service/jenkinsApi.ts (hunk 1 above)
Comment (timja) [reply plugins/jenkins-backend/src/service/jenkinsApi.ts:180]: It doesn’t compile without it

Comment (timja) [reply plugins/jenkins-backend/src/service/jenkinsApi.ts:180]: We are using the latest version.

> "@types/jenkins@npm:^0.23.1":
  version: 0.23.3

Comment (timja) [reply plugins/jenkins-backend/src/service/jenkinsApi.ts:180]: types mustn't have been updated for the 1.0.0 release, https://github.com/backstage/backstage/issues/13692#issuecomment-1249593608

https://github.com/silas/node-jenkins/compare/0.28.1...1.0.1

The pre-existing comment one line above said there was already an issue with the types, I removed the `as any` from the end of the line, but had to add it the constructor to make it work

Comment (benjdlambert) [reply plugins/jenkins-backend/src/service/jenkinsApi.ts:180]: Hmm - these types are really not even useful to have in the project at this point to be honest. Wondering if we should roll out own or maybe contribute type fixes. Happy probably get this through in the meantime, but it's a little unfortunate that they haven't been updated. :)

**Review by benjdlambert — APPROVED** [review]

**Comment by benjdlambert** [comment]
Thanks for this :pray:

**Comment by planeiii-te** [comment]
It seems like this upgrade creates a `tsc` error:

```shell
➜  backstage git:(upgrade-jenkins-plugin) ✗ gd

packages/app/package.json

────┐21: │
────┘│ 21 │    "@backstage/plugin-catalog-import": "^0.8.11",                                                                                          │ 21 │    "@backstage/plugin-catalog-import": "^0.8.11",
│ 22 │    "@backstage/plugin-catalog-react": "^1.1.3",                                                                                            │ 22 │    "@backstage/plugin-catalog-react": "^1.1.3",
│ 23 │    "@backstage/plugin-github-actions": "^0.5.8",                                                                                           │ 23 │    "@backstage/plugin-github-actions": "^0.5.8",
│ 24 │    "@backstage/plugin-jenkins": "^0.7.8",                                                                                                  │ 24 │    "@backstage/plugin-jenkins": "^0.7.7",                                                                                                  
│ 25 │    "@backstage/plugin-org": "^0.5.8",                                                                                                      │ 25 │    "@backstage/plugin-org": "^0.5.8",
│ 26 │    "@backstage/plugin-permission-react": "^0.4.4",                                                                                         │ 26 │    "@backstage/plugin-permission-react": "^0.4.4",
│ 27 │    "@backstage/plugin-scaffolder": "^1.5.0",                                                                                               │ 27 │    "@backstage/plugin-scaffolder": "^1.5.0",

Binary files a/yarn.lock and b/yarn.lock differ

packages/app/package.json
➜  backstage git:(upgrade-jenkins-plugin) ✗ yarn tsc
yarn run v1.22.19
$ tsc
packages/backend/src/plugins/jenkins.ts:20:7 - error TS2741: Property 'validateEntity' is missing in type 'CatalogClient' but required in type 'CatalogApi'.

20       catalog,
         ~~~~~~~

  node_modules/@backstage/plugin-jenkins-backend/node_modules/@backstage/catalog-client/dist/index.d.ts:336:5
    336     validateEntity(entity: Entity, location: string, options?: CatalogRequestOptions): Promise<ValidateEntityResponse>;
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    'validateEntity' is declared here.
  node_modules/@backstage/plugin-jenkins-backend/dist/index.d.ts:78:9
    78         catalog: CatalogApi;
               ~~~~~~~
    The expected type comes from property 'catalog' which is declared here on type '{ config: Config; catalog: CatalogApi; }'

Found 1 error in packages/backend/src/plugins/jenkins.ts:20error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by timja** [comment]
What makes you think it is this PR?

**Comment by timja** [comment]
I upgraded just fine in https://github.com/hmcts/backstage/pull/85, no issues

**Comment by benjdlambert** [comment]
How did you upgrade? Did you run `yarn backstage-cli versions:bump`? looks like there's type issue somewhere and duplicate versions of the `catalog-client` in the `jenkins-backend` plugin. You could maybe delete that version of the `catalog-client` from the yarn.lock and run yarn again to fix it. But running the `versions:bump` command should remove duplicates

**Comment by planeiii-te** [comment]
I posted this before I realized there were several PRs to this plugin. I just upgraded via the command line:

```shell
➜  backstage git:(upgrade-jenkins-plugin) yarn add --cwd packages/backend @backstage/plugin-jenkins-backend
```

The fix was simple:

```shell
➜  backstage git:(upgrade-jenkins-plugin) yarn add --cwd packages/backend @backstage/catalog-client
 ```

**Pull request merged**